src/Commands/Extension/AzdCommand.cs

```diff
@@ -89,7 +93,60 @@ public override async Task<CommandResponse> ExecuteAsync(CommandContext context,
             });
 
             var azdPath = FindAzdCliPath() ?? throw new FileNotFoundException("Azure Developer CLI executable not found in PATH or common installation locations. Please ensure Azure Developer CLI is installed.");
-            var result = await processService.ExecuteAsync(azdPath, command, _processTimeoutSeconds);
+
+            // Check if this command should use real-time streaming output
+            bool shouldStreamOutput = ShouldUseRealtimeStreaming(command);
+            ProcessResult result;
+            
+            if (shouldStreamOutput)
+            {
+                // Create a StringBuilder to collect the streamed output for the final result
+                var streamedOutput = new List<string>();
+                
+                // Define handlers for real-time output
+                void StdOutHandler(string line)
+                {
+                    streamedOutput.Add(line);
+                    
+                    // Send the output line as it arrives
+                    var streamingResponse = new CommandResponse
```

Comment:
I may be mistaken but the MCP protocol doesn't currently contain a concept of partial results or streaming.  When you return the first result, the MCP client will think that is the response from calling the tool.

Reply:
It does https://mcp-framework.com/docs/Transports/http-stream-transport

Reply:
> It does

That isn't about streaming the results of tools; see @halter73's comment here:
https://github.com/modelcontextprotocol/csharp-sdk/issues/358#issuecomment-2832612965